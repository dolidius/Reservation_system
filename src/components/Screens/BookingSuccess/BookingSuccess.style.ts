import styled from "styled-components";

export const Container = styled.div`
    width: 100%;
    height: 100%;
    padding: 3rem 7rem;
`

export const Title = styled.h1`
    font-size: 3.5rem;
    color: white;
`

export const TitleSecondary = styled.h2`
    font-size: 2rem;
    color: white;
`

export const SeatTypography = styled.p`
    color: white;
    font-size: 1.3rem;
    letter-spacing: 1px;
`