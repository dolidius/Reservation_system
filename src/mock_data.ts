export const seats =  [
    {
      "id": "s02",
      "cords": {
        "x": 0,
        "y": 2
      },
      "reserved": false
    },
    {
      "id": "s03",
      "cords": {
        "x": 0,
        "y": 3
      },
      "reserved": false
    },
    {
      "id": "s04",
      "cords": {
        "x": 0,
        "y": 4
      },
      "reserved": false
    },
    {
      "id": "s06",
      "cords": {
        "x": 0,
        "y": 6
      },
      "reserved": false
    },
    {
      "id": "s07",
      "cords": {
        "x": 0,
        "y": 7
      },
      "reserved": false
    },
    {
      "id": "s08",
      "cords": {
        "x": 0,
        "y": 8
      },
      "reserved": true
    },
    {
      "id": "s09",
      "cords": {
        "x": 0,
        "y": 9
      },
      "reserved": true
    },
    {
      "id": "s011",
      "cords": {
        "x": 0,
        "y": 11
      },
      "reserved": false
    },
    {
      "id": "s012",
      "cords": {
        "x": 0,
        "y": 12
      },
      "reserved": false
    },
    {
      "id": "s013",
      "cords": {
        "x": 0,
        "y": 13
      },
      "reserved": false
    },
    {
      "id": "s014",
      "cords": {
        "x": 0,
        "y": 14
      },
      "reserved": false
    },
    {
      "id": "s12",
      "cords": {
        "x": 1,
        "y": 2
      },
      "reserved": false
    },
    {
      "id": "s13",
      "cords": {
        "x": 1,
        "y": 3
      },
      "reserved": false
    },
];