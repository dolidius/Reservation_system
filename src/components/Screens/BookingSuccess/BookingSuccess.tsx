import React from 'react';

import { connect } from 'react-redux';
import IBookingState from '../../../interfaces/storeStates/IBookingState';
import { ViewportContainer } from '../../../styles/Layout/Container.style';

import { Title, Container, TitleSecondary, SeatTypography } from './BookingSuccess.style';

import { Box } from '@material-ui/core';

interface IProps {
    bookingStore: IBookingState;
}

const BookingSuccess: React.FC<IProps> = ({ bookingStore }) => {
    return (
        <ViewportContainer>
            <Container>
                <Title>
                    Twoja rezerwacja przebiegła pomyślnie!
                </Title>

                <Box mt={5}>
                    <TitleSecondary>
                        Wybrałeś miejsca:
                    </TitleSecondary>
                </Box>

                {bookingStore.bookedSeats.map(seat => (
                    <SeatTypography key={`s${seat.x}${seat.y}`}>
                        -rząd {seat.x}, miejsce {seat.y}, s{`${seat.x}${seat.y}`}
                    </SeatTypography>
                ))}

                <Box mt={7}>
                    <TitleSecondary>
                        Dziękujemy! W razie problemów prosimy o kontakt z działem administracji
                    </TitleSecondary>
                </Box>

            </Container>
        </ViewportContainer>
    )
}

const mapStateToProps = (state: any) => ({
    bookingStore: state.booking
})

export default connect(mapStateToProps, {})(BookingSuccess);