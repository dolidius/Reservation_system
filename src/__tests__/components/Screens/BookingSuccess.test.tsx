import BookingSuccess from '../../../components/Screens/BookingSuccess/BookingSuccess';
import { Provider } from "react-redux";

import * as renderer from "react-test-renderer";

import { bookedSeats } from '../../../mock_data';

import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

const middlewares = [thunk];

const mockStore = configureStore(middlewares);

let store = mockStore({
    booking: {
        bookedSeats
    }
});

describe("<BookingSuccess />", () => {

    it("should render correctly given seats data from store", () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <BookingSuccess />
                </Provider>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

})