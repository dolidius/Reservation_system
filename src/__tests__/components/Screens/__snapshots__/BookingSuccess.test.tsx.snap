// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookingSuccess /> should render correctly given booked seats data from store 1`] = `
<div
  className="sc-gtsrHT eltPMO"
>
  <div
    className="sc-dlnjwi elroaH"
  >
    <h1
      className="sc-hKFxyN hVUWNH"
    >
      Twoja rezerwacja przebiegła pomyślnie!
    </h1>
    <div
      className="MuiBox-root MuiBox-root-1"
    >
      <h2
        className="sc-eCApnc gtqReh"
      >
        Wybrałeś miejsca:
      </h2>
    </div>
    <p
      className="sc-jSFjdj jASire"
    >
      -rząd 
      1
      , miejsce 
      2
      , s
      12
    </p>
    <p
      className="sc-jSFjdj jASire"
    >
      -rząd 
      1
      , miejsce 
      3
      , s
      13
    </p>
    <p
      className="sc-jSFjdj jASire"
    >
      -rząd 
      3
      , miejsce 
      4
      , s
      34
    </p>
    <p
      className="sc-jSFjdj jASire"
    >
      -rząd 
      0
      , miejsce 
      5
      , s
      05
    </p>
    <div
      className="MuiBox-root MuiBox-root-2"
    >
      <h2
        className="sc-eCApnc gtqReh"
      >
        Dziękujemy! W razie problemów prosimy o kontakt z działem administracji
      </h2>
    </div>
  </div>
</div>
`;

exports[`<BookingSuccess /> should render correctly given seats data from store 1`] = `
<div
  className="sc-gtsrHT eltPMO"
>
  <div
    className="sc-dlnjwi elroaH"
  >
    <h1
      className="sc-hKFxyN hVUWNH"
    >
      Twoja rezerwacja przebiegła pomyślnie!
    </h1>
    <div
      className="MuiBox-root MuiBox-root-1"
    >
      <h2
        className="sc-eCApnc gtqReh"
      >
        Wybrałeś miejsca:
      </h2>
    </div>
    <p
      className="sc-jSFjdj jASire"
    >
      -rząd 
      1
      , miejsce 
      2
      , s
      12
    </p>
    <p
      className="sc-jSFjdj jASire"
    >
      -rząd 
      1
      , miejsce 
      3
      , s
      13
    </p>
    <p
      className="sc-jSFjdj jASire"
    >
      -rząd 
      3
      , miejsce 
      4
      , s
      34
    </p>
    <p
      className="sc-jSFjdj jASire"
    >
      -rząd 
      0
      , miejsce 
      5
      , s
      05
    </p>
    <div
      className="MuiBox-root MuiBox-root-2"
    >
      <h2
        className="sc-eCApnc gtqReh"
      >
        Dziękujemy! W razie problemów prosimy o kontakt z działem administracji
      </h2>
    </div>
  </div>
</div>
`;
